### ISSUE TITLE
Need for scaling

### ISSUE THREAD

**Issue by ariG23498** [keras-team/keras-io#655]
Hey @AakashKumarNain I love the Image Captioning example. There is just one thing I would like to ask here.

Why do we scale the embeddings prior to adding the positional encoding?
https://github.com/keras-team/keras-io/blob/26424c7544511f4a9c7464d2018a6ae2d350f058/examples/vision/image_captioning.py#L345

I have not seen the explanation of this bit anywhere. 

CC: @fchollet

**Comment by SuryanarayanaY** [comment]
Hi @ariG23498 ,

It is mentioned in original transformer [paper](https://arxiv.org/pdf/1706.03762.pdf) to scale the embedding inputs to sqrt(model_dimension). Hence the implementation is like that. 

<img width="949" alt="Screenshot 2023-08-02 at 3 51 26 PM" src="https://github.com/keras-team/keras-io/assets/116063290/60db859e-f110-4c98-b94e-a24ddeb16aac">


Maybe It's kind of hyperparameter that was found effective during research. Since in original transformers the decoder output is offset by one position and its coming through softmax activation layer scaling up these values may avoid vanishing gradient problems.Since without scaling the word embeddings are small values(probabilities) and some of them might be even tiny during initial phase of training and these are going to added to positional encodings there might be chance of loss of information. Even though there is no concrete answer in paper the author of tutorial seems followed the paper and implemented in that way.Hope this will helps some extent. Thanks!

**Comment by ariG23498** [comment]
Thanks @SuryanarayanaY !
